### ISSUE TITLE
SFTP user chroot shell path update & ls addition

### ISSUE THREAD

**Issue by phillxnet** [rockstor/rockstor-core#2863]
During #2858's development, it was noted that we have the following shell setup enforced for SFTP users:

https://github.com/rockstor/rockstor-core/blob/ce442438aedad79edf242e84cc764642572edc74/src/rockstor/system/ssh.py#L253

I.e. we modify the SFTP  user's shell to `/bin/bash` which is likely now inconsistent with the new chroot environment setup by way of SFTP configuration.

---

Reproducer of consequent behaviour to follow.

**Comment by phillxnet** [comment]
Post having enabled sftp we have the following config for the specific user (owner of the share we sftp export):
via:
> cat /etc/ssh/sshd_config

```
###BEGIN: Rockstor SFTP CONFIG. DO NOT EDIT BELOW THIS LINE###
Subsystem       sftp    internal-sftp
AllowUsers root sftp-user
Match User sftp-user
        ForceCommand internal-sftp
        ChrootDirectory /mnt3/sftp-user
        X11Forwarding no
        AllowTcpForwarding no

```
With a necessary minor edit (remarking out) above this re the sftp subsystem line:

> #Subsystem      sftp    /usr/lib/ssh/sftp-server

**Comment by phillxnet** [comment]
Once we setup a proposed `sftp-user`, and a SFTP share owned by them, i.e. `sftp-share` and enable the export of this share we have the setup used during the final testing of #2858 (https://github.com/rockstor/rockstor-core/pull/2858#issuecomment-2196599533):

> As per documented setup:
> "Secure File Transport Protocol (SFTP)"
> https://rockstor.com/docs/interface/storage/file_sharing/sftp.html
> 
> 1. create sftp-user
> 2. create sftp-share
> 3. Edit sftp-share Access control to owner=sftp-user
> 4. Add SFTP Share sftp-share

If we then attempt to ssh in as this user:
```
ssh sftp-user@rleap15-6.lan
Password: 
```
The result we receive is, as per our "ForceCommand internal-sftp" configuration":
```
This service allows sftp connections only.
Connection to rleap15-6.lan closed.
```

**Comment by phillxnet** [comment]
So given the last comment, our now inconsistent bash path, given the changes to our chroot introduced in #2858, has no current artifacts as we "ForceCommand" the internal-sftp as the user 'shell'. However we may want to accommodate custom behaviour in the future that utilises our chroot setup. Doing a quick investigation of this now.

**Comment by phillxnet** [comment]
By way of testing a future modification re no or custom ForceCommand, if we remove the ForceCommand line, and stop/start all rockstor* services; then wipe/re-enable the same sftp-share export we end-up with the following config:
```
###BEGIN: Rockstor SFTP CONFIG. DO NOT EDIT BELOW THIS LINE###
Subsystem       sftp    internal-sftp
AllowUsers root sftp-user
Match User sftp-user
        ChrootDirectory /mnt3/sftp-user
        X11Forwarding no
        AllowTcpForwarding no
```

**Comment by phillxnet** [comment]
If we correct the sftp-user shell path, stop/start all `rockstor*` services and exercise our proposed custom sftp/sshconfig with the ForceCommand line removed (restoring a chroot bash shell) we have a successful ssh login for the same sftp-user:
```
ssh sftp-user@rleap15-6.lan
Password: 
Have a lot of fun...
Last login: Mon Jul  8 18:26:30 2024 from 192.168.2.172
-bash-4.4$
```
This is a highly constrained shell as we have only bash and rsync binaries (plus lib dependencies) accessible: but that is outside the scope of this issue. I.e.:
```
-bash-4.4$ ls
-bash: ls: command not found
-bash-4.4$
```

**Comment by phillxnet** [comment]
By way of providing at least `ls` if we add this to our `PROGS_IN_CHROOT` , along with our path  correction, we can see at least our chroot's `\` thus:
```
ssh sftp-user@rleap15-6.lan
Password: 
Have a lot of fun...
Last login: ...
-bash-4.4$ ls -la
total 0
drwxr-xr-x 1    0 0  42 Jul  8 17:48 .
drwxr-xr-x 1    0 0  42 Jul  8 17:48 ..
drwxr-xr-x 1    0 0   0 Jun 28 17:01 lib
drwxr-xr-x 1    0 0 182 Jun 28 10:23 lib64
drwxr-xr-x 1 1006 0   0 Jul  8 16:05 sftp-share
drwxr-xr-x 1    0 0  16 Jun 28 10:23 usr
```
Assuming the proposed user intervention of removing our overarching constraint of the "ForcedCommand" limitation for all SFTP users.

**Comment by phillxnet** [comment]
Closing as:
Fixed by #2865
